### DISCUSSION TITLE
Add custom endpoints to SDK

### DISCUSSION THREAD

**Discussion by juancarlosjr97** [directus/directus#4696] [Feature Requests]
What do you think of adding a method to allow the SDK to call endpoints?

**Comment by barakplasma** [comment]
I also would like a method for calling endpoints (ala trpc if possible).
Our workaround at the moment is reusing the directus transport [docs](https://docs.directus.io/reference/sdk.html#customize-transport) (which is an instance of the [axios api](https://axios-http.com/docs/post_example)) which is already authenticated to directus and using it as the baseUrl.
```js
const relativeDirectusEndpoint = "/misc/function-endpoint"
const requestBody = {foo: 1}
const response = await directus.transport.post(relativeDirectusEndpoint, requestBody);
const returned = response.raw
```

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
